client/lib/posts/post-edit-store.js

```diff
@@ -134,7 +139,15 @@ function set( attributes ) {
 		_queue.push( attributes );
 	}
 
-	updatedPost = assign( {}, _post, attributes );
+	let updatedPost = assignWith( clone( _post ), attributes, ( objValue, srcValue, key ) => {
```

Comment:
is this usage of `assignWith` the equivalent to the following code:

```js
let updatedPost = { 
  ..._post, 
  ...attributes, 
  metadata: mergeMetadataEdits( _post.metadata, attributes.metadata ) 
};
```

Reply:
ah, it isn't because `metadata` doesn't always exist.
I still think we can have a much simpler solution albeit it may need mutation. 
Heres an example:

```js
	let updatedPost = { ..._post, ...attributes };
if ( _post.metadata && attributes.metadata ) {
  updatedPost.metadata = mergeMetadataEdits( _post.metadata, attributes.metadata );
}
```

Reply:
Good point, I got so carried away by fancy Lodash functions to completely forget that the object spread operator can do the task equally well 😀 

```js
if ( _post.metadata && attributes.metadata ) {
  updatedPost.metadata = mergeMetadataEdits( _post.metadata, attributes.metadata );
}
```
I only removed the `if ( _post.metadata )` condition which means "if the post doesn't have any metadata yet, ignore the edits".

Also, I changed `mergeMetadataEdits` to cope with possible `null` value of existing metadata, just to be safe. Achieved by using `_.filter` instead of default fn argument and `Array.prototype.filter`.